packages/mui-lab/src/internal/pickers/hooks/usePickerState.ts

```diff
@@ -95,6 +102,7 @@ export function usePickerState<TInput, TDateValue>(
       onClear: () => acceptDate(valueManager.emptyValue, true),
       onAccept: () => acceptDate(draftState.draft, true),
       onDismiss: () => setIsOpen(false),
+      onCancel: () => acceptDate(initialDate as unknown as TDateValue, true),
```

Comment:
Do you want to call `acceptDate` here? That kicks off `onChange` and `onAccept` which may cause unintended behavior. Perhaps a new method which closes the dialog and resets draftState

Reply:
I see your point, but I think this is good enough for now.